Assets/MixedRealityToolkit/_Core/Interfaces/Sharing/ISimpleState.cs

```diff
@@ -0,0 +1,15 @@
+﻿using System.Threading.Tasks;
+
+namespace Microsoft.MixedReality.Toolkit.Core.Interfaces.Sharing
+{
+    /// <summary>
+    /// Simplest possible non-automatic state sharing.
+    /// Will typically be networked but could also plug into a file system or local database.
+    /// </summary>
+    public interface ISimpleState : IMixedRealityService
+    {
+        Task<bool> HasKey(string key);
+        Task SetState<T>(string key, T state);
+        Task<T> GetState<T>(string key);
```

Comment:
Do we expect that most users will just want to obtain state values on each frame update? Would it be more convenient to use subscriptions/callbacks on value updates here compared to only having a getter?

Reply:
Most likely so. Any implementation should cache stuff locally so we're not reaching across the network for stale data. But this process should be invisible. Callbacks are probably a notch or two more sophisticated than our needs in this case.